dpnp/dpnp_algo/dpnp_algo_mathematical.pyx

```diff
@@ -498,25 +499,25 @@ cpdef dparray dpnp_sum(dparray input, axis=None):
     return dpnp_result_array
 
 
-cpdef dpnp_trapz(dparray y, dparray x, int dx):
+cpdef dpnp_trapz(dparray y1, dparray x1, double dx):
 
-    len = y.size
+    if y1.size <= 1:
+        if y1.dtype == dpnp.float32:
+            return dpnp.array([0], dtype=dpnp.float32)
+        return dpnp.array([0], dtype=dpnp.float64)
```

Comment:
remove this if this case implemented on backend, or just move this logic on backend.

Leave only getting funcs pointers and mem alloc for dparrays on cython level.